coreblocks/utils/utils.py

```diff
@@ -219,3 +220,11 @@ def has_explicit_shape(val: ValueLike):
                     "Shapes not matching: lhs: {} {} rhs: {} {}".format(lhs.shape(), lhs, rhs_val.shape(), rhs)
                 )
         yield lhs.eq(rhs_val)
+
+
+def layout_subset(layout: LayoutList, *, fields: set[str]) -> LayoutList:
+    result = []
+    for field in layout:
+        if field[0] in fields:
+            result.append(field)
+    return result
```

Comment:
Why not use a list comprehension?
```suggestion
    return [item for item in layout if item[0] in fields]
```

Reply:
I guess I haven't written enough Python code recently :)